ompi/mca/pml/ucx/pml_ucx.c

```diff
@@ -697,13 +697,8 @@ int mca_pml_ucx_isend_init(const void *buf, size_t count, ompi_datatype_t *datat
     req->send.mode                = mode;
     req->send.ep                  = ep;
     req->ompi_datatype            = datatype;
-
-    if (MCA_PML_BASE_SEND_BUFFERED == mode) {
-        OBJ_RETAIN(datatype);
-    } else {
-        req->datatype = mca_pml_ucx_get_datatype(datatype);
```

Comment:
i think this is still needed because bsend does not use UCX datatype (it packs to temp buffer using opal convertor)
e.g
if (mode == BUFFERED) {
  req->datatype = NULL;
} else {
  req->datatype = mca_pml_ucx_get_datatype(datatype);
}